### PR TITLE
CXSPA-4006: refactor for using captcha

### PR DESCRIPTION
**Pull request by scarai-sap**

### CODE REVIEW

File: projects/storefrontlib/shared/components/captcha/captcha.model.ts [hunk 1]
```diff
@@ -10,11 +10,6 @@ import { CaptchaConfig } from '@spartacus/core';
 
 export interface RenderParams {
   element?: HTMLElement | string;
-  /**
-   * Public key
-   * In order to render the reCaptcha widget. Get the reCaptcha token using publicKey when rendering the widget.
-   */
-  publicKey?: string;
```
Comment (i53577) [inline projects/storefrontlib/shared/components/captcha/captcha.model.ts:17]: as a sample, maybe it's better to keep `publicKey`?

File: projects/storefrontapp-e2e-cypress/cypress/e2e/regression/user_access/mockCaptcha.core-e2e.cy.ts [hunk 2]
```diff
@@ -35,10 +35,7 @@ describe('Register', () => {
       });
       cy.findByText(/Sign in \/ Register/i).click();
 
-      cy.intercept('GET', /\.*\/bframe\?hl=.*/).as('getIFrame');
       cy.get('cx-login-register').findByText('Register').click();
-      cy.wait('@getIFrame').its('response.statusCode').should('eq', 200);
-      cy.get('cx-captcha').find('iframe').should('exist');
```
Comment (i53577) [inline projects/storefrontapp-e2e-cypress/cypress/e2e/regression/user_access/mockCaptcha.core-e2e.cy.ts]: keep this to ensure captcha element has been rendered.
```
cy.get('cx-captcha').should('exist')
```

File: projects/storefrontlib/shared/components/captcha/mockCaptcha/config/captcha-api-config.ts [hunk 3]
```diff
@@ -13,12 +13,10 @@ import { CaptchaProvider } from '../../captcha.model';
   providedIn: 'root',
   useExisting: Config,
 })
-export abstract class GoogleRecaptchaApiConfig {
-  apiUrl?: string;
-  fields?: { [key: string]: string };
```
Comment (i53577) [inline projects/storefrontlib/shared/components/captcha/mockCaptcha/config/captcha-api-config.ts]: same question, as a sample, is it better to keep `apiUrl` and `fields`?

File: projects/storefrontapp-e2e-cypress/cypress.config.ci.ts [hunk 4]
```diff
@@ -29,9 +29,6 @@ export default defineConfig({
     OCC_PREFIX: '/occ/v2',
     OCC_PREFIX_USER_ENDPOINT: 'users',
     OCC_PREFIX_ORDER_ENDPOINT: 'orders',
-    RECAPTCHA_PUBLIC_KEY: '6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI',
```
Comment (i333055) [inline projects/storefrontapp-e2e-cypress/cypress.config.ci.ts:32]: do we need delete BACK_OFFICE_URL and BACK_OFFICE_PREFIX?

**Pull request closed without merging**